Algorithm.Framework/Selection/OpenInterestFutureUniverseSelectionModel.cs

```diff
@@ -0,0 +1,138 @@
+            {
+                throw new ArgumentNullException(nameof(algorithm));
+            }
+
+            _algorithm = algorithm;
+            _resultsLimit = resultsLimit;
+            _chainContractsLookupLimit = chainContractsLookupLimit;
+            _exchangeHours = SecurityExchangeHours.AlwaysOpen(TimeZones.Utc);
+        }
+
+        /// <inheritdoc cref="FutureUniverseSelectionModel.Filter" />
+        protected override FutureFilterUniverse Filter(FutureFilterUniverse filter)
+        {
+            return filter.Contracts(FilterByOpenInterest(filter));
+        }
+
+        /// <summary>
+        ///     Filters a set of contracts based on open interest.
+        /// </summary>
+        /// <param name="contracts">Contracts to filter</param>
+        /// <returns>Filtered set</returns>
+        public IEnumerable<Symbol> FilterByOpenInterest(IEnumerable<Symbol> contracts)
+        {
+            return contracts.OrderBy(x => x.ID.Date).Take(_chainContractsLookupLimit).OrderByDescending(GetOpenInterest).ThenBy(x => x.ID.Date).Take(_resultsLimit);
+        }
+
+        private decimal GetOpenInterest(Symbol symbol)
+        {
+            var current = _algorithm.Time;
+            var previousBusinessDate = GetPreviousBusinessDay(_algorithm.Time, symbol);
+            var result = _algorithm.HistoryProvider.GetHistory(
+                    new[]
+                    {
+                        new HistoryRequest(
+                            previousBusinessDate,
+                            current,
+                            typeof(Tick),
+                            symbol,
+                            Resolution.Tick,
+                            _exchangeHours,
+                            TimeZones.Utc,
+                            null,
+                            true,
+                            false,
+                            DataNormalizationMode.Raw,
+                            TickType.OpenInterest
+                        )
+                    },
+                    TimeZones.Utc
+                )
+                .FirstOrDefault();
+
+            var ticks = result?.Ticks?[symbol];
+            if (ticks == null || ticks.Count == 0)
+            {
+                _algorithm.Log($"WARN: Unable to retrieve open interest for: {symbol}");
+                return 0;
+            }
+
+            if (ticks.Count != 1)
+            {
+                var message = new StringBuilder(nameof(OpenInterestFutureUniverseSelectionModel)).Append('.')
+                    .Append(MethodBase.GetCurrentMethod().Name)
+                    .Append("(): Received ")
+                    .Append(ticks.Count)
+                    .Append(" entries for ")
+                    .Append(symbol.ToString())
+                    .Append(" tick data containing open interest. Will use first. Values: ")
+                    .Append(string.Join(",", ticks));
+                _algorithm.Error(message.ToString());
```

Comment:
Not sure this is an error? The history call could return more than 1 data point using tick resolution

Reply:
Removed